### PR TITLE
Add support for scs-eureka

### PR DESCRIPTION
**Pull request by dyroberts**
- Extract autoconfig (moved to new scs autoconfig modules)
 - Add `CfEurekaClientProcessor`
connected to #66

### CODE REVIEW

**Comment by spencergibb** [comment]
Porting auto config seems out of character for this project. Seems like it should just get properties into the correct for so auto config in some other project could take advantage. If some auto config that was previously in SCS connectors but wanted to move to pure OSS, seems like spring-cloud-netflix would be a better fit for that.

File: java-cfenv-boot-pivotal-scs/src/main/java/io/pivotal/cfenv/boot/scs/config/CfSpringCloudConfigClientProcessor.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.pivotal.cfenv.boot.scs;
+package io.pivotal.cfenv.boot.scs.config;
```
Comment (markpollack) [inline java-cfenv-boot-pivotal-scs/src/main/java/io/pivotal/cfenv/boot/scs/config/CfSpringCloudConfigClientProcessor.java]: While moving to a config package is nicer, it isn't the kind of change to make in a point release (which I am thinking where we will add this functionality).

**Comment by dyroberts** [comment]
@spencergibb I can see the logic in what you're saying. In this case, I was following the pattern set by the scs config-server support, which includes some auto-config. In fact, I think the main reason there is an scs-specific artifact, here in this project, is to provide this auto-config. Maybe you could log an issue to move all of this to spring-cloud-netflix?

File: java-cfenv-boot-pivotal-scs/src/main/java/io/pivotal/cfenv/boot/scs/config/CfSpringCloudConfigClientProcessor.java (hunk 1 above)
Comment (dyroberts) [reply java-cfenv-boot-pivotal-scs/src/main/java/io/pivotal/cfenv/boot/scs/config/CfSpringCloudConfigClientProcessor.java]: Ok, that makes sense. I'll move it all into one big happy package! :)

**Comment by markpollack** [comment]
> Porting auto config seems out of character for this project. Seems like it should just get properties into the correct for so auto config in some other project could take advantage. If some auto config that was previously in SCS connectors but wanted to move to pure OSS, seems like spring-cloud-netflix would be a better fit for that.

That does sound reasonable, but while `EurekaInstanceAutoConfiguration` will not kick-in unless it is an app running on CF, 

```@Configuration
@ConditionalOnProperty("eureka.client.serviceUrl.defaultZone")
@ConditionalOnExpression("'${vcap.application.uris[0]:}'!='' || '${cf.instance.ip:}'!=''")
public class EurekaInstanceAutoConfiguration {
```

The class `EurekaOAuth2AutoConfiguration` looks like it would need some additional conditionals to similarly restrict when it kicks in?

Also, there are a couple of existing autoconfig classes in the released code
```
org.springframework.boot.autoconfigure.EnableAutoConfiguration=\
io.pivotal.cfenv.boot.scs.VaultTokenRenewalAutoConfiguration,\
io.pivotal.cfenv.boot.scs.PlainTextConfigClientAutoConfiguration
``` 

Looking into it now, I can't actually figure out how `PlainTextConfigClientAutoConfiguration` is being used since it produced a bean of the interface type `PlainTextConfigClient` and that doesn't seem to be used anywhere.

**Comment by markpollack** [comment]
There is also `SanitizingEurekaInstanceConfigBean` in this autoconfig which may be something only needed in CF environments?

**Comment by spencergibb** [comment]
Didn't notice the config server stuff.

**Comment by spencergibb** [comment]
Glancing at config stuff, I'd say the same thing. Why do auto config here?

**Comment by markpollack** [comment]
Would be a good opportunity to fill in the class level Javadoc with some description....

**Comment by markpollack** [comment]
I don't see what `PlainTextConfigClientAutoConfiguration` is actually doing anyway...... maybe it is cruft moved over?

**Comment by dyroberts** [comment]
@markpollack @spencergibb Regarding the auto-configuration, scs has historically provided its own starter library, as well as connector. Maybe it makes more sense for that to continue, replacing the connector with java-cfenv?

**Comment by dyroberts** [comment]
Looking at the `EurekaOAuth2AutoConfiguration` and `ConfigClientOAuth2BootstrapConfiguration`, it feels like the oauth2 portion of the auto-config could be moved to spring-cloud-netflix and spring-cloud-config respectively.

**Comment by dyroberts** [comment]
@markpollack @spencergibb I've added a commit to this pr to start using the new version of spring-security, as defined by the spring-boot bom.

**Comment by dyroberts** [comment]
@markpollack @spencergibb As discussed I've removed all autoconfig from this library and moved it to a new module that will be included, along with this library, in the scs starters.

**Comment by markpollack** [comment]
merged in 738960b4224d59818faa1fab95ce10f5c8cb2e2d

**Comment by xyloman** [comment]
@markpollack do you know when the 1.1.2.RELEASE will be cut that contains these updates?  I have had a few teams recently ask about the support and they need this capability in order to move off of spring cloud connectors.

If you were looking for testers before moving over we might be able to carve off some time to help test.

**Comment by markpollack** [comment]
Dylan and I made a release the 1.1.2 release a few min ago, give it a whirl.

**Comment by xyloman** [comment]
@markpollack we took these updates for a spin and with the removal of EurekaOAuth2AutoConfiguration and ConfigClientOAuth2BootstrapConfiguration how would these properties be leveraged that java-cfenv-boot-pivotal-scs sets?

**Comment by dyroberts** [comment]
Hi @xyloman. Sorry for the confusion. The official spring-cloud-services client libraries are migrating to java-cfenv and the AutoConfig etc has been moved there where it ultimately belongs. There should be a new release out later this week that will depend on java-cfenv-boot-pivotal-scs:1.1.2.RELEASE. After that, you'll be able to follow the docs on connecting client apps (https://docs.pivotal.io/spring-cloud-services/3-1/common/client-dependencies.html). For now, you can continue to use the previous java-cfenv release.

**Pull request closed without merging**